src/test/java/com/commercetools/sync/services/impl/CustomObjectServiceImplTest.java

```diff
@@ -0,0 +1,246 @@
+        when(result.head()).thenReturn(Optional.of(mock));
+
+        when(client.execute(any())).thenReturn(CompletableFuture.completedFuture(result));
+
+
+        final Optional<CustomObject<JsonNode>> customObjectOptional = service
+            .fetchCustomObject(CustomObjectCompositeIdentifier.of(customObjectKey, customObjectContainer))
+            .toCompletableFuture().join();
+
+        assertAll(
+            () -> assertThat(customObjectOptional).containsSame(mock),
+            () -> assertThat(
+                service.keyToIdCache.get(
+                    CustomObjectCompositeIdentifier.of(customObjectKey, customObjectContainer).toString())
+            ).isEqualTo(customObjectId)
+        );
+        verify(client).execute(any(CustomObjectQuery.class));
+
+
+    }
+
+    @Test
+    void createCustomObject_WithDraft_ShouldCreateCustomObject() {
+        final CustomObject mock = mock(CustomObject.class);
+        when(mock.getId()).thenReturn(customObjectId);
+        when(mock.getKey()).thenReturn(customObjectKey);
+        when(mock.getContainer()).thenReturn(customObjectContainer);
+
+        when(client.execute(any())).thenReturn(CompletableFuture.completedFuture(mock));
+
+        final ObjectNode customObjectValue = JsonNodeFactory.instance.objectNode();
+        customObjectValue.put("currentHash", "1234-5678-0912-3456");
+        customObjectValue.put("convertedAmount", "100");
+
+
+        final CustomObjectDraft<JsonNode> draft = CustomObjectDraft
+                .ofUnversionedUpsert(customObjectContainer, customObjectKey,customObjectValue);
+
+        final Optional<CustomObject<JsonNode>> customObjectOptional =
+                service.upsertCustomObject(draft).toCompletableFuture().join();
+
+        assertThat(customObjectOptional).containsSame(mock);
+        verify(client).execute(eq(CustomObjectUpsertCommand.of(draft)));
+    }
+
+    @Test
+    void createCustomObject_WithRequestException_ShouldNotCreateCustomObject() {
+        final CustomObject mock = mock(CustomObject.class);
+        when(mock.getId()).thenReturn(customObjectId);
+
+        when(client.execute(any())).thenReturn(CompletableFutureUtils.failed(new BadRequestException("bad request")));
+
+        final CustomObjectDraft<JsonNode> draftMock = mock(CustomObjectDraft.class);
+        when(draftMock.getKey()).thenReturn(customObjectKey);
+        when(draftMock.getContainer()).thenReturn(customObjectContainer);
+        when(draftMock.getJavaType()).thenReturn(getCustomObjectJavaTypeForValue(convertToJavaType(JsonNode.class)));
+
+        final Optional<CustomObject<JsonNode>> customObjectOptional =
+            service.upsertCustomObject(draftMock).toCompletableFuture().join();
+
+        String expectedMsg = "Failed to create draft with key: '{key='%s', container='%s'}'";
+        assertAll(
+            () -> assertThat(customObjectOptional).isEmpty(),
+            () -> assertThat(errorMessages).hasOnlyOneElementSatisfying(message -> {
```

Comment:
I see `hasOnlyOneElementSatisfying` as deprecated - can you use non-deprecated version?

Reply:
/commits/4394d8ed3927c801344a04aaea5693d44fc8949d